lib/spack/spack/test/conftest.py

```diff
@@ -683,3 +685,32 @@ def invalid_spec(request):
     """Specs that do not parse cleanly due to invalid formatting.
     """
     return request.param
+
+#######
+# Mock Configure executable
+#######
+
+
+@pytest.fixture(scope="function")
+def mock_configure():
```

Comment:
A couple requests:

* move this fixture to the architecture tests module since it is only used there
* return the value of `MODULE_LOADED`